src/core/metadata/qgslayermetadataproviderregistry.cpp

```diff
@@ -0,0 +1,69 @@
+/***************************************************************************
+  qgslayermetadataproviderregistry.cpp - QgsLayerMetadataProviderRegistry
+
+ ---------------------
+ begin                : 17.8.2022
+ copyright            : (C) 2022 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgslayermetadataproviderregistry.h"
+#include "qgsabstractlayermetadataprovider.h"
+#include "qgsfeedback.h"
+
+QgsLayerMetadataProviderRegistry::QgsLayerMetadataProviderRegistry( QObject *parent ) : QObject( parent )
+{
+
+}
+
+void QgsLayerMetadataProviderRegistry::registerLayerMetadataProvider( QgsAbstractLayerMetadataProvider *metadataProvider )
+{
+  mMetadataProviders.insert( metadataProvider->type(), metadataProvider );
+}
+
+void QgsLayerMetadataProviderRegistry::unregisterLayerMetadataProvider( QgsAbstractLayerMetadataProvider *metadataProvider )
+{
+  delete mMetadataProviders.take( metadataProvider->type() );
+}
+
+QList<QgsAbstractLayerMetadataProvider *> QgsLayerMetadataProviderRegistry::layerMetadataProviders() const
+{
+  return mMetadataProviders.values();
+}
+
+QgsAbstractLayerMetadataProvider *QgsLayerMetadataProviderRegistry::layerMetadataProviderFromType( const QString &type )
+{
+  return mMetadataProviders.value( type, nullptr );
+}
+
+QgsLayerMetadataSearchResult QgsLayerMetadataProviderRegistry::search( const QString &searchString, const QgsRectangle &geographicExtent, QgsFeedback *feedback )
+{
+  QgsLayerMetadataSearchResult results;
+  for ( auto it = mMetadataProviders.cbegin(); it != mMetadataProviders.cend(); ++it )
+  {
+
+    if ( feedback && feedback->isCanceled() )
+    {
+      break;
+    }
+
+    const QgsLayerMetadataSearchResult providerResults { it.value()->search( searchString, geographicExtent ) };
+    const QList<QgsLayerMetadataProviderResult> cMetadata { providerResults.metadata() };
```

Comment:
what's the "c" in "cMetadata" represent?

Reply:
const, it's used widely in the rest of the code

Reply:
I'm -0 to introducing a new prefix like this -- without widespread use it's not really telling us anything new (and even here there's a bunch of `std::as_const( cSomething ) ` calls, which suggest the hint isn't working as intended 🤪

Reply:
This was actually discussed sometimes, IIRC @m-kuhn suggested to leave std::as_const even if the container is already const. So, my standard m.o. is to first create a const copy of the container, prefix it with `c` and use std::as_const.

Would you prefer `constMetadata` instead?

Reply:
> Would you prefer constMetadata instead?

I'm -0 to introducing a new prefix in a piecemeal fashion, unless we can do some kind of "bulk rename" (eg via a clang tidy tool). Ie it's not the prefix I'm opposed to, it's just having it used only in a tiny tiny % of QGIS code alone.